### PR TITLE
quantlib: Initial integration

### PR DESCRIPTION
**Pull request by silvergasp**

### CODE REVIEW

**Comment by silvergasp** [comment]
TODO:
- [ ] Fill in contact details
- [ ] Switch from my personal fork to mainline

**Comment by silvergasp** [comment]
cc @lballabio

File: projects/quantlib/build.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+mkdir build
+cd build
+export LDFLAGS="-Wl,-rpath,'\$ORIGIN/lib'"
+cmake .. -GNinja -DBOOST_ROOT=/usr -DCMAKE_BUILD_TYPE=Release -DQL_COMPILE_WARNING_AS_ERROR=ON -L
```
Comment (lballabio) [inline projects/quantlib/build.sh:20]: You'll need to enable `QL_BUILD_FUZZ_TEST_SUITE` now that it's off by default, I guess?

Comment (silvergasp) [reply projects/quantlib/build.sh:20]: Oh yeah, that'd be why it's not working cheers. That's an easy enough fix.

File: projects/quantlib/project.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+homepage: "https://www.quantlib.org/"
+language: c++
+# TODO: Get primary author dets
+primary_contact: ""
```
Comment (silvergasp) [inline projects/quantlib/project.yaml:4]: @lballabio is it ok if I put the gmail account listed on your github profile here, or would you prefer a different contact?

Comment (lballabio) [reply projects/quantlib/project.yaml:4]: The address of the QuantLib users mailing list would be best.

Comment (silvergasp) [reply projects/quantlib/project.yaml:4]: Got a couple of quetions about that;
- Are you sure? this email will receive bug-reports about potential security vulnerabilities. So if it get's sent to a public mailing list all vulnerabilities will be immediately public.
- Is the mailing list email address a gmail/google account, [it is one of the requirements for getting access to the oss-fuzz dashboard](https://google.github.io/oss-fuzz/faq/#why-do-you-require-a-google-account-for-authentication).

Comment (lballabio) [reply projects/quantlib/project.yaml:4]: Ok, use my gmail address then.

Comment (silvergasp) [reply projects/quantlib/project.yaml]: done

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Pull request merged**